test/hooks.test.js

```diff
@@ -740,23 +740,27 @@ test('onRoute hook should able to change the route url', t => {
   })
 })
 
-test('onRoute hook that throws should be caught', t => {
+test('onRoute hook that throws should be caught', { only: true }, t => {
   t.plan(1)
   const fastify = Fastify({ exposeHeadRoutes: false })
 
   fastify.register((instance, opts, done) => {
     instance.addHook('onRoute', () => {
       throw new Error('snap')
     })
-    instance.get('/', opts, function (req, reply) {
-      reply.send()
-    })
+
+    try {
+      instance.get('/', opts, function (req, reply) {
+        reply.send()
+      })
```

Comment:
Same here. `t.fail` with a declarative description.

Reply:
Done in 0849c1b 👍